src/diffusion_for_multi_scale_molecular_dynamics/models/axl_diffusion_lightning_model.py

```diff
@@ -417,11 +454,46 @@ def _get_coordinates_target_normalized_score(
         delta_relative_coordinates = map_relative_coordinates_to_unit_cell(
             noisy_relative_coordinates - real_relative_coordinates
         )
-        target_normalized_scores = get_sigma_normalized_score(
+        target_normalized_scores = get_coordinates_sigma_normalized_score(
             delta_relative_coordinates, sigmas, kmax=self.hyper_params.kmax_target_score
         )
         return target_normalized_scores
 
+    def _get_lattice_target_normalized_score(
+        self,
+        noisy_lattice_parameters: torch.Tensor,
+        real_lattice_parameters: torch.Tensor,
+        sigmas_n: torch.Tensor,
+        alpha_bars: torch.Tensor,
+    ) -> torch.Tensor:
+        """Get target normalized score for the lattice parameters.
+
+        It is assumed that the inputs are consistent, ie, the noisy lattice parameters correspond
+        to the real lattice parameters noised with sigmas and alpha bars. It is also assumed that sigmas has
+        been broadcast so that the same value sigma(t) is applied to all atoms + dimensions within a configuration.
+
+        # TODO add angles
+
+        Args:
+            noisy_lattice_parameters : noised lattice parameters.
+                Tensor of dimensions [batch_size, spatial_dimension]
+            real_lattice_parameters : original lattice coordinates, before the addition of noise.
+                Tensor of dimensions [batch_size, spatial_dimension]
+            sigmas_n : variance scaled by the number of atoms
+                Tensor of dimensions [batch_size, spatial_dimension]
+            alpha_bars :
+                Tensor of dimensions [batch_size, spatial_dimension]
+
+        Returns:
+            target normalized score: sigma times target score, ie, sigma times nabla_lt log P_{t|0}(lt| l0).
+                Tensor of dimensions [batch_size, spatial_dimension]
+        """
+        delta_relative_coordinates = noisy_lattice_parameters - real_lattice_parameters
```

Comment:
I'm not sure this is right.

Looking at Eq. (A7) from MatterGen,
<img width="520" alt="Screenshot 2024-12-23 at 4 25 57 PM" src="https://github.com/user-attachments/assets/1c7cf238-83e1-4c73-8ee1-7ecb8ac42128" />

the reference position is not "x_0", it's something modified by "sqrt alpha bar". I looked at the  method ` get_lattice_sigma_normalized_score`: it doesn't account for this transformation.

Reply:
good catch - this is done in the score function as suggested in another comment